server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -222,39 +224,60 @@ public boolean parsesArrayValue() {
     public void parse(DocumentParserContext context) throws IOException {
         try {
             if (hasScript) {
-                throw new IllegalArgumentException("Cannot index data directly into a field with a [script] parameter");
+                throwIndexingWithScriptParam();
```

Comment:
NIT: I would not extract this unless absolutely necessary. Not sure if this is saving a lot method size wise...